src/Sarif/Core/PropertyBagHolder.cs

```diff
@@ -217,10 +218,7 @@ public void SetPropertiesFrom(IPropertyBagHolder other)
 
         public void RemoveProperty(string propertyName)
         {
-            if (Properties != null)
-            {
-                Properties.Remove(propertyName);
-            }
+                Properties?.Remove(propertyName);
```

Comment:
It looks like the design is settled here, but I wonder if we should've thrown in this case. Callers could try to clean up properties on the wrong objects in the SARIF hierarchy and not realize nothing is actually happening underneath.

Reply:
It's a good point. Changing this behavior would of course be breaking (not that that's ever stopped us before ;-)).

---
In reply to: [329663092](https://github.com/microsoft/sarif-sdk/pull/1688#discussion_r329663092) [](ancestors = 329663092)